packages/mui-material/src/Menu/Menu.js

```diff
@@ -307,6 +328,12 @@ Menu.propTypes /* remove-proptypes */ = {
     PropTypes.func,
     PropTypes.object,
   ]),
+  /**
+   * The component used for the transition.
+   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
+   * @default Grow
+   */
+  TransitionComponent: PropTypes.elementType,
```

Comment:
Why are we adding this one? `slots.transition` should be used instead.

Reply:
I think it's because I move `TransitionComponent` to the the props, let me find other way.

Reply:
But did it work beofre this? I'm asking because it's not included in the API: https://mui.com/material-ui/api/menu/.

If it never worked before, there's no need to add it now.

If it worked, we can add it but we should:
- Add it to the types (as it was incorrectly missing before)
- Add the deprecated message

Reply:
> If it never worked before, there's no need to add it now.

This is because the `TransitionComponent` pass through `…other` before. But now, we need this backward compatibility with `TransitionComponent={slots.transition ?? other.TransitionComponent}`.

The moment I added the line, the `generatePropTypes` attached `TransitionComponent` to the proptypes. I haven't find a way to disable it in this case.

I think I will work on Popover first (to deprecate the `TransitionComponent` and then once it's done will revisit this PR)

Reply:
Oh, I see now.

Right, so if we finish the Popover first, then we can use its `slots` instead of `TransitionComponent` and we don't need to rescue `TransitionComponent` from `other`, this plan makes sense 👍🏼

Reply:
With the latest change, no more `TransitionComponent` changes.